### PR TITLE
ROC-3113: Use new payment api for pba account

### PR DESCRIPTION
**Pull request by sabahirfan**
This PR implements the integration with new payment api, to submit payment by PBA account before issuing a claim.

**Jira:** [ROC-3113](https://tools.hmcts.net/jira/browse/ROC-3113)

### CODE REVIEW

File: src/main/app/pay/model/paymentResponse.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  readonly reference: string
+
+  @Expose({ name: 'date_created' })
+  readonly dateCreated: string
+
+  readonly status: string
+
+  get isSuccess (): boolean {
+    return this.status === 'Success' || this.status === 'Initiated'
```
Comment (yarpo) [inline src/main/app/pay/model/paymentResponse.ts]: Maybe let's create enum for these values?

File: config/custom-environment-variables.json [hunk 2]
```diff
@@ -39,6 +39,9 @@
   "fees": {
     "url": "FEES_URL"
   },
+  "pay": {
```
Comment (timja-kainos) [inline config/custom-environment-variables.json]: missing pay healthcheck

File: config/default.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "service-2-service-auth": {
       "url": "http://localhost:4552",
       "totpSecret": "AAAAAAAAAAAAAAAA",
-      "microservice": "cmc_legal_frontend"
+      "microservice": "cmc"
```
Comment (timja-kainos) [inline config/default.json]: need to change the secret in `main.tf` to use the cmc one

File: src/main/features/claim/routes/pay-by-account.ts [hunk 4]
```diff
@@ -106,6 +120,21 @@ export default express.Router()
...
+
+        const paymentResponse: PaymentResponse = await payClient.create(
+          res.locals.user,
+          draft.document.feeAccount.reference,
+          draft.document.externalId,
+          draft.document.yourReference.reference,
+          feeResponse
+        )
+
+        if (paymentResponse.isSuccess) {
```
Comment (timja-kainos) [inline src/main/features/claim/routes/pay-by-account.ts]: jira seems to say this should be done after saving the case

File: config/default.json (hunk 3 above)
Comment (sabahirfan) [reply config/default.json]: done

File: src/main/features/claim/routes/pay-by-account.ts (hunk 4 above)
Comment (sabahirfan) [reply src/main/features/claim/routes/pay-by-account.ts]: Jira needs to be updated.

File: config/custom-environment-variables.json (hunk 2 above)
Comment (sabahirfan) [reply config/custom-environment-variables.json]: done

**Pull request merged**